### PR TITLE
Fastq to fasta fix

### PR DESCRIPTION
**Pull request by neoformit**
Fix `fastx_toolkit/fastq_to_fasta`. Resolves https://github.com/galaxyproject/tools-iuc/issues/5366

* [x] - This PR updates an existing tool or tool collection

- Add a select input so the user can specify whether they want compressed FASTA output [ `infer` (default), `yes`, `no` ].
- Add some tests

### CODE REVIEW

**Comment by neoformit** [comment]
@bernt-matthias

**Comment by bernt-matthias** [comment]
Yes? Guess tool test and linting problems need to be fixed :)

**Comment by neoformit** [comment]
Sorry should have checked those... my planemo was out of date :(

File: tool_collections/fastx_toolkit/fastq_to_fasta/fastq_to_fasta.xml [hunk 1]
```diff
@@ -9,7 +9,11 @@
 $SKIPN
 $RENAMESEQ
 -v @FQQUAL@
-@GZIP@ > '$output'
+#if $compress_output == 'yes' or $compress_output == 'infer' and $input.ext.endswith('.gz') or $input.ext.endswith('.bz2'):
```
Comment (bernt-matthias) [inline tool_collections/fastx_toolkit/fastq_to_fasta/fastq_to_fasta.xml:12]: Should be fine by evaluation order, but I think it makes it more readable, but up to you

```suggestion
#if $compress_output == 'yes' or ($compress_output == 'infer' and ($input.ext.endswith('.gz') or $input.ext.endswith('.bz2'))):
```

**Review by bernt-matthias — APPROVED** [review]

File: tool_collections/fastx_toolkit/fastq_to_fasta/fastq_to_fasta.xml [hunk 2]
```diff
@@ -49,45 +45,45 @@ $RENAMESEQ
...
+            <output name="output" file="fastq_to_fasta-out2.out" expect_num_outputs="1" />
         </test>
         <test>
             <!-- 6 Force FASTQ-To-FASTA.GZ -->
             <param name="input" value="fastq_to_fasta-in2.fastq" ftype="fastqsanger" />
             <param name="SKIPN" value="no"/>
             <param name="RENAMESEQ" value="yes"/>
             <param name="compress_output" value="yes"/>
-            <output name="output" file="fastq_to_fasta-out2.out.gz" />
+            <output name="output" file="fastq_to_fasta-out2.out.gz" expect_num_outputs="1" />
```
Comment (bgruening) [inline tool_collections/fastx_toolkit/fastq_to_fasta/fastq_to_fasta.xml]: is that correct? expect_num_outputs should be on the top `<test expect_num_outputs="1">` level.

**Review by bgruening — APPROVED** [review]

File: tool_collections/fastx_toolkit/fastq_to_fasta/fastq_to_fasta.xml (hunk 2 above)
Comment (bernt-matthias) [reply tool_collections/fastx_toolkit/fastq_to_fasta/fastq_to_fasta.xml]: Nah.

Comment (bernt-matthias) [reply tool_collections/fastx_toolkit/fastq_to_fasta/fastq_to_fasta.xml]: I would have expected that the xsd linter raises an error...

Comment (neoformit) [reply tool_collections/fastx_toolkit/fastq_to_fasta/fastq_to_fasta.xml]: Yeah that was just me being half asleep on the job, think I fixed this up?

**Comment by bernt-matthias** [comment]
@neoformit @bgruening just realized that this broke the other tools of the fastx_toolkit (due to the symlinked macros file). https://github.com/galaxyproject/tools-iuc/issues/5412

How to proceed?

**Comment by neoformit** [comment]
I'll revert the linked macro file back to prev version and replace the linked macros.xml in fastq_to_fasta with the updated/fixed version. That at least keeps everything working for now. I haven't checked whether the faulty logic in fastq_to_fasta can also be found across the suite, and should perhaps be fixed globally?

**Comment by neoformit** [comment]
Actually I just reverted to the old linked macros.xml as it's forwards compatible with my changes to fastq_to_fasta.

**Pull request merged**